src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -419,7 +428,7 @@ class RedownloadService[F[_]: NonEmptyParallel](
       _ <- logger.debug("Removing unaccepted snapshots from disk.")
       _ <- removeUnacceptedSnapshotsFromDisk().value.flatMap(F.fromEither)
 
-      _ <- if (shouldPerformRedownload) {
+      _ <- if (shouldPerformRedownload && !isDownload) { // I think we should only set it to Ready when we are not joining
```

Comment:
correct. is it a code that is being executed for download and redownloads? or redownloads only?

Reply:
it executed for both (so it set the node as Ready before logic in def download did it), now it will only execute for redownloads.